src/HtsSystemContract.sol

```diff
@@ -213,14 +213,17 @@ contract HtsSystemContract is IHederaTokenService, IERC20Events {
             emit Approval(owner, spender, amount);
             return abi.encode(true);
         } else if (selector == IHRC719.associate.selector) {
+            _initTokenRelationships(msg.sender);
```

Comment:
This should be similar to `_balanceOfSlot`, maybe we can override `_isAssociatedSlot` and the the mechanisms used there.